src/server/Agent.js

```diff
@@ -31,6 +31,10 @@ export default class Agent extends SideAgent {
         this._actualListeningAddress = null; // used (and fully resolved)
 
         this._transaction = new Transaction(this);
+
+        this._serverClosed = null; // a promise to close this.server
+
+        this._serverSocket = null; // server transaction socket
```

Comment:
I bet most readers will see serverSocket and think it is a listening socket!

```suggestion
        // the first accepted (or inherited) connection
        this._firstConnectionSocket = null;
```

We should not  call it firstTransactionSocket or, better, store it inside `_transaction` because there may be _multiple_ transactions on this connection, right?